Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/parsers/BaseCurrencyParser.java

```diff
@@ -77,13 +79,16 @@ private ParseResult mergeCompoundUnit(ExtractResult compoundResult) {
                 result = new ParseResult(extractResult.getStart(), extractResult.getLength(), extractResult.getText(), extractResult.getType(), null, null, null);
 
                 mainUnitValue = unitValue;
-                numberValue = parseResultValue.isPresent() ? Double.parseDouble(parseResultValue.get().number) : 0;
+                if (parseResultValue.isPresent() && parseResultValue.get().number != null) {
+                    numberValue = Double.parseDouble(parseResultValue.get().number);
+                }
+
                 result.setResolutionStr(parseResult.getResolutionStr());
                 mainUnitIsoCode = config.getCurrencyNameToIsoCodeMap().containsKey(unitValue) ? config.getCurrencyNameToIsoCodeMap().get(unitValue) : mainUnitIsoCode;
 
                 // If the main unit can't be recognized, finish process this group.
                 if (mainUnitIsoCode == null || mainUnitIsoCode.isEmpty()) {
-                    result.setValue(new UnitValue(String.valueOf(numberValue), mainUnitValue));
+                    result.setValue(new UnitValue(numberValue < 0 ? null : getNumberValue(numberValue), mainUnitValue));
```

Comment:
This check for the invalid number could be moved inside getNumberValue() and improved to check for exactly the new constant value.